libtransmission/utils.mm

```diff
@@ -0,0 +1,37 @@
+// This file Copyright © 2022-2023 Mnemosyne LLC.
+// It may be used under GPLv2 (SPDX: GPL-2.0-only), GPLv3 (SPDX: GPL-3.0-only),
+// or any future license endorsed by Mnemosyne LLC.
+// License text can be found in the licenses/ folder.
+
+#import <Foundation/Foundation.h>
+#include <string>
+#include <string_view>
+#include <utf8.h>
+
+// macOS implementation of tr_strv_replace_invalid() that autodetects the encoding.
+// This replaces the generic implementation of the function in utils.cc
+
+std::string tr_strv_replace_invalid(std::string_view sv, uint32_t replacement)
+{
+    // UTF-8 encoding
+    NSString* validUTF8 = @(std::string(sv).data());
```

Comment:
If we have to have the std::string temporary here, `.c_str()` is slightly preferable since it's const:

```suggestion
    NSString* validUTF8 = @(std::string(sv).c_str());
```

Reply:
`c_str()` makes a copy and `@()` makes a copy too. So `data()` is best for this particular case.

Reply:
Either way is fine here and equivalent both in terms of performance & results so if you feel strongly I'm not going to reject the PR or anything :smile_cat: 

But to put on my pedantic hat -- `c_str()` doesn't make a copy; [c_str() and data() perform the same function](https://en.cppreference.com/w/cpp/string/basic_string/c_str) but c_str() is more idiomatic since it's always `const`. The extra copy is made when we instantiate the `std::string` which is why I was wondering about `initWithBytes`